pkg/status/statuscollector.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func (c *Controller) syncStatusCollector(ctx context.Context, key string) error {
+
+	statusCollector, err := c.statusCollectorLister.Get(key)
+	if err != nil {
+		// The resource no longer exist, which means it has been deleted.
+		if errors.IsNotFound(err) {
+			return nil
```

Comment:
If the StatusCollector no longer exists, the resolver should still be updated, in order to delete its occurrences in resolutions. I'll update API.